zerver/lib/message.py

```diff
@@ -254,7 +255,15 @@ def messages_for_ids(
 
     for message_id in message_ids:
         msg_dict = message_dicts[message_id]
-        msg_dict.update(flags=user_message_flags[message_id])
+        flags = user_message_flags[message_id]
+        # For clients which don't support the renamed `stream_wildcard_mentioned`
+        # flag, we send the `wildcard_mentioned` flag. All clients will be
+        # required to support `stream_wildcard_mentioned` flag in the future;
+        # this logic is intended for backwards-compatibility only.
+        if not stream_wildcard_mentioned_flag and "stream_wildcard_mentioned" in flags:
+            flags.remove("stream_wildcard_mentioned")
+            flags.append("wildcard_mentioned")
```

Comment:
This may be a good question for the "api design" thread -- I wonder if it'd be better to __always__ send `stream_wildcard_mentioned`, and just have sending the `wildcard_mentioned` copy of it be the thing that's conditional on the client capability.

(The benefit of that plan would be that it'd make it easier to delete the client capability in 3 years, once there's no longer supported clients that don't support the new format).